src/main/java/fr/sncf/osrd/train/TrainState.java

```diff
@@ -277,6 +277,9 @@ private void step(
     public Train.TrainStateChange evolveStateUntilTime(Simulation sim, double targetTime) {
         var locationChange = new Train.TrainStateChange(sim, trainSchedule.trainID, this);
 
+        if (targetTime - this.time < 1e-5)
+            return locationChange;
+
```

Comment:
Maybe the fix should be in `step`, the function shouldn't fail it the time step is `0`